package/MDAnalysis/converters/RDKit.py

```diff
@@ -325,9 +330,9 @@ def convert(self, obj, cache=True, NoImplicit=True, max_iter=200,
                 # assign coordinates
                 conf = Chem.Conformer(mol.GetNumAtoms())
                 for atom in mol.GetAtoms():
-                    idx = atom.GetIntProp("_MDAnalysis_index")
+                    idx = atom.GetIdx()
```

Comment:
Just commenting to think out loud -- this is fine because `_MDAnalysis_index` is set sequentially in `atomgroup_to_mol`.